pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/blocks/TransferableBlockUtils.java

```diff
@@ -19,33 +19,28 @@
 package org.apache.pinot.query.runtime.blocks;
 
 import org.apache.pinot.common.utils.DataSchema;
-import org.apache.pinot.common.utils.DataTable;
-import org.apache.pinot.core.common.datablock.BaseDataBlock;
 import org.apache.pinot.core.common.datablock.DataBlockUtils;
+import org.apache.pinot.core.common.datablock.MetadataBlock;
 
 
 public final class TransferableBlockUtils {
   private TransferableBlockUtils() {
     // do not instantiate.
   }
-  private static final TransferableBlock EOS_TRANSFERABLE_BLOCK =
-      new TransferableBlock(DataBlockUtils.getEndOfStreamDataBlock());
 
-  public static TransferableBlock getEndOfStreamTransferableBlock() {
-    return EOS_TRANSFERABLE_BLOCK;
+  public static TransferableBlock getEndOfStreamTransferableBlock(DataSchema dataSchema) {
+    return new TransferableBlock(DataBlockUtils.getEmptyDataBlock(dataSchema));
   }
 
   public static TransferableBlock getErrorTransferableBlock(Exception e) {
     return new TransferableBlock(DataBlockUtils.getErrorDataBlock(e));
   }
 
-  public static TransferableBlock getEmptyTransferableBlock(DataSchema dataSchema) {
-    return new TransferableBlock(DataBlockUtils.getEmptyDataBlock(dataSchema));
+  public static TransferableBlock repackErrorBlock(MetadataBlock upstreamErrorBlock) {
+    return new TransferableBlock(DataBlockUtils.getErrorDataBlock(upstreamErrorBlock.getExceptions()));
```

Comment:
Why can't we return the error block as is to downstream operators ?

Reply:
oh because i need to repack with the current operator's dataschema

Reply:
echo back to this point from https://github.com/apache/pinot/pull/9064#discussion_r928010027, we don't need to repackage if we don't include schema into the mix in error blocks